### PR TITLE
Add 3p vulkan validation layer support. 

### PR DESCRIPTION
**Pull request by moudgils**
Enabling device validation will automatically load the VkLayer_khronos_validation.dll ensuring that validation checks will work correctly

Signed-off-by: moudgils <47460854+moudgils@users.noreply.github.com>

### CODE REVIEW

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/Instance.cpp [hunk 1]
```diff
@@ -45,8 +46,22 @@ namespace AZ
 
         bool Instance::Init(const Descriptor& descriptor)
         {
-            m_descriptor = descriptor;   
+            m_descriptor = descriptor;
+            if (GetValidationMode() != RHI::ValidationMode::Disabled)
+            {
+                //This env var (VK_LAYER_PATH) is used by the drivers to look for VkLayer_khronos_validation.dll
+                AZStd::string exeFolder = AZ::Test::GetCurrentExecutablePath().c_str();
```
Comment (jeremyong-az) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/Instance.cpp:53]: Nit: `const char* exeFolder` saves you a string copy.

File: Gems/Atom/RHI/Vulkan/Code/Source/RHI/Instance.cpp [hunk 2]
```diff
@@ -45,8 +46,22 @@ namespace AZ
...
+                AZ::Test::SetEnv("VK_LAYER_PATH", exeFolder.c_str(), 1);
 
+                RawStringList validationLayers = Debug::GetValidationLayers();
+                m_descriptor.m_optionalLayers.insert(m_descriptor.m_requiredLayers.end(), validationLayers.begin(), validationLayers.end());
+                m_descriptor.m_optionalExtensions.push_back(VK_EXT_DEBUG_REPORT_EXTENSION_NAME);
+                m_descriptor.m_optionalExtensions.push_back(VK_EXT_DEBUG_UTILS_EXTENSION_NAME);
```
Comment (jeremyong-az) [inline Gems/Atom/RHI/Vulkan/Code/Source/RHI/Instance.cpp:59]: Is there an issue if the debug utils extension is requested twice in this list?

**Review by jeremyong-az — APPROVED** [review]
Minor comments but approved otherwise

**Review by spham-amzn — APPROVED** [review]

**Comment by moudgils** [comment]
> Should we not move the function from test into some core Az location?
> AzTest usually requires linking to the aztest library (or is that all part of azcore anyway?)

I am all for moving SetEnv to AzCore. In fact I was surprised that it was in AzTest but I don't want to do that as part of this work. I can update it in future if this ever gets moved. AzCore does have test code which includes AzTest but it may just be for AzCoreTest related libraries. AzTest is quite small so it seems ok to link.

**Pull request merged**